### PR TITLE
Route type declaration

### PR DESCRIPTION
**Pull request by A-ndy-git**
Hello,

Middleware package `@middy/http-router` has no input type defined for `httpRouterHandler`.

When attempting to use the HTTP Router middleware in Typescript, you receive an error `Expected 0 arguments, but got 1.ts(2554)`.

This PR adds a new `Route` interface for `httpRouterHandler`.

### CODE REVIEW

**Comment by willfarrell** [comment]
Thanks for the PR! Can you update the tests so they pass please. I should be able to get this released once it passes CI.

File: packages/http-router/index.test-d.ts [hunk 1]
```diff
@@ -1,6 +1,10 @@
-import { expectType } from 'tsd'
-// import middy from '@middy/core'
-import httpRouterHandler from '.'
+import middy from '@middy/core';
+import httpRouterHandler from '.';
+import { expectType } from 'tsd';
 
-const middleware = httpRouterHandler()
-expectType<any>(middleware)
+const middleware = validator({
```
Comment (willfarrell) [inline packages/http-router/index.test-d.ts:5]: This should be `const middleware = httpRouterHandler({`

**Review by willfarrell — CHANGES_REQUESTED** [review]

File: packages/http-router/index.test-d.ts (hunk 1 above)
Comment (A-ndy-git) [reply packages/http-router/index.test-d.ts]: Sorry @willfarrell. First commit to this repo, tried to do it on the fly with [github.dev](url).

Hopefully all fixed now and for future PRs I'll revert back to the IDE :)

**Comment by willfarrell** [comment]
FYI you can run `npm run test:packages:typings` and `npm run test:lint:ci` locally to test.

**Comment by A-ndy-git** [comment]
Going to resubmit this under a new PR to mitigate the extra conversation here.

**Pull request closed without merging**